engines/ospray/ispc/geometry/SDFBeziers.ispc

```diff
@@ -0,0 +1,444 @@
+/* Copyright (c) 2015-2018, EPFL/Blue Brain Project
+ * All rights reserved. Do not distribute without permission.
+ * Author: Sebastien Speierer <sebastien.speierer@epfl.ch>
+ * *
+ * This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+// ospray
+#include "ospray/SDK/common/Model.ih"
+#include "ospray/SDK/common/Ray.ih"
+#include "ospray/SDK/geometry/Geometry.ih"
+#include "ospray/SDK/math/box.ih"
+#include "ospray/SDK/math/vec.ih"
+// embree
+#include "embree3/rtcore.isph"
+#include "embree3/rtcore_geometry.isph"
+#include "embree3/rtcore_scene.isph"
+
+#include "ospray/SDK/math/vec.ih"
+
+#include "utils/SafeIncrement.ih"
+
+#include "RayMarching.isph"
+
+//////////////////////////////////////////////////////////////////////
+
+struct SDFBezier
+{
+    uint64 userData;
+    vec3f p0; // starting point
+    vec3f c0; // first control point
+    float r0; // starting radius
+    vec3f p1; // ending point
+    vec3f c1; // second control point
+    float r1; // ending radius
+};
+
+DEFINE_SAFE_INCREMENT(SDFBezier);
+
+//////////////////////////////////////////////////////////////////////
+
+inline float mix(const float x, const float y, const float a)
+{
+    return x * (1.f - a) + y * a;
+}
+
+inline uniform float bezier(uniform float t, uniform float p0, uniform float c0,
+                            uniform float p1, uniform float c1)
+{
+    const uniform float s = 1 - t;
+    const uniform float s2 = s * s;
+    const uniform float t2 = t * t;
+    return s * s2 * p0 + 3 * s2 * t * c0 + 3 * s * t2 * c1 + t * t2 * p1;
+}
+
+inline vec3f bezier(float t, const uniform vec3f& p0, const uniform vec3f& c0,
+                    const uniform vec3f& p1, const uniform vec3f& c1)
+{
+    const vec3f s = make_vec3f(1 - t);
+    const vec3f s2 = s * s;
+    const vec3f t2 = make_vec3f(t * t);
+    return s * s2 * p0 + 3 * s2 * t * c0 + 3 * s * t2 * c1 + t * t2 * p1;
+}
+
+//////////////////////////////////////////////////////////////////////
+
+inline float sdf_tube(const vec3f& P, const vec3f& P0, const vec3f& P1,
+                      float r0_, float r1_, float t, float deltaT)
+{
+#define TANGENTIAL_TUBE
```

Comment:
What is the difference between these two methods, in terms of performance and visual quality?